qiskit/algorithms/aux_ops_evaluator.py

```diff
@@ -0,0 +1,157 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021, 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""Evaluator of auxiliary operators for algorithms."""
+
+from typing import Tuple, Union, List
+
+import numpy as np
+
+from qiskit import QuantumCircuit
+from qiskit.algorithms.minimum_eigen_solvers.minimum_eigen_solver import ListOrDict
+from qiskit.opflow import (
+    CircuitSampler,
+    ListOp,
+    StateFn,
+    CircuitStateFn,
+    OperatorBase,
+    ExpectationBase,
+)
+from qiskit.providers import BaseBackend, Backend
+from qiskit.utils import QuantumInstance
+
+
+def eval_observables(
+    quantum_instance: Union[QuantumInstance, BaseBackend, Backend],
+    ansatz: QuantumCircuit,
```

Comment:
Would this be better to be called `state` now since this is the state for which we want the expectation value. Yes it is a result from say an ansatz elsewhere like vqe, but at this point its a known specific quantity. The text describing this can then say circuit (no parameters) representing the state for which expectation values....

If we say no parameters do we want to check here too and raise an error - otherwise some error will happen later but it may be less evident.